app/scripts/models/WorkspaceResolver.js

```diff
@@ -31,7 +31,7 @@ define(['backbone', 'FLOOD'],
       console.log(this.workspace.get('name') + " has dependencies: " + JSON.stringify( depIds) );
 
       var that = this;
-      this.app.get('workspaces').on('add', function(ws){ that.resolveDependency.call(that, ws); }, this);
+      this.listenTo(this.app.get('workspaces'), 'add', this.resolveDependency);
 
```

Comment:
This is not bound correctly.  Remember that the workspace collection (first arg) is expected to be `this` in the function (third arg).  The original code was too conservative, but your fix looks incorrect.